chain/beacon/sync_manager.go

```diff
@@ -91,32 +101,41 @@ func (s *SyncManager) Stop() {
 
 type requestInfo struct {
 	nodes []net.Peer
+	from  uint64
 	upTo  uint64
 }
 
 // RequestSync asks the sync manager to sync up with those peers up to the given
 // round. Depending on the current state of the syncing process, there might not
 // be a new process starting (for example if we already have the round
-// requested). upTo == 0 means the syncing process goes on forever.
-func (s *SyncManager) RequestSync(nodes []net.Peer, upTo uint64) {
+// requested). upTo == 0 means the syncing process goes on forever. from == 0 means
+// we start from the latest stored beacon. from > 0 means we force-sync.
+func (s *SyncManager) RequestSync(from, upTo uint64, nodes []net.Peer) {
 	s.newReq <- requestInfo{
 		nodes: nodes,
+		from:  from,
 		upTo:  upTo,
 	}
 }
 
-func (s *SyncManager) Run() {
+func (s *SyncManager) Run(ctx context.Context) {
 	// no need to sync until genesis time
 	for s.clock.Now().Unix() < s.info.GenesisTime {
 		time.Sleep(time.Second)
 	}
 	// tracks the time of the last round we successfully synced
 	lastRoundTime := 0
 	// the context being used by the current sync process
-	lastCtx, cancel := context.WithCancel(context.Background()) // nolint
+	lastCtx, cancel := context.WithCancel(ctx)
 	for {
 		select {
 		case request := <-s.newReq:
+			if request.from > 0 {
+				// we always do it and we block while doing it if it's a forced sync.
+				s.log.Debugw("Starting forced sync", "from", request.from, "upTo", request.upTo)
+				s.Sync(lastCtx, request)
```

Comment:
Can you explain a bit more the behaviour here ? 
Two questions:
* Shouldn't we `cancel()` as well ? Otherwise that means there are two sync in parallel and that's the reason syncmanager was created in the first place, to avoid this.
* Why blocking ? Why you call this "force" ?

Reply:
This was moved in the end.